### PR TITLE
[Placement Group] Support Placement Group state table.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This partially complete https://github.com/ray-project/ray/issues/9802. Example output:
```bash
ray.experimental.placement_group_table(pid)

{'placement_group_id': 'fd858c4c4fff86a9a55174c16845af9c', 'name': 'name', 'bundles': {0: {'CPU': 2.0}, 1: {'CPU': 2.0}}, 'strategy': 'SPREAD', 'state': 'ALIVE'}
```
This is useful for debugging / testing placement group, so I implemented the partial feature ahead of time. This will be used in testing https://github.com/ray-project/ray/pull/10063.

### NOTE
- It doesn't have cpp tests yet because it requires to add a AsyncAdd method which won't be used except tests. It will be handled in the future. 
- Java APIs are also not supported. @ffbin I will collect all placement group APIs to another issue later, so please add them to Java accordingly.
- This somehow formats some files. I am using clang-format 7.0, so it should be the right formatting (lmk if it is not).

## Related issue number

https://github.com/ray-project/ray/issues/9802

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/gcs_utils.py [hunk 1]
```diff
@@ -1,48 +1,19 @@
...
-    TablePrefix,
-    TablePubsub,
-    TaskTableData,
-    ResourceMap,
-    ResourceTableData,
-    ObjectLocationInfo,
-    PubSubMessage,
-    WorkerTableData,
-)
+    ActorCheckpointIdData, ActorTableData, GcsNodeInfo, JobTableData,
```
Comment (ffbin) [inline python/ray/gcs_utils.py:2]: How about leave the format unchanged?

File: python/ray/state.py [hunk 2]
```diff
@@ -377,6 +377,54 @@ def profile_table(self):
...
+                placement_group_info = (gcs_utils.PlacementGroupTableData.
+                                        FromString(placement_group_info))
+                return self._gen_placement_group_info(placement_group_info)
+        else:
+            raise NotImplementedError(
+                "Get all placement group is not implemented yet.")
+
+    def _gen_placement_group_info(self, placement_group_info):
+        def get_state(state):
+            if state == 0:
```
Comment (ffbin) [inline python/ray/state.py:402]: How about define `PlacementGroupState` in gcs_utils.py? just like `ray.gcs_utils.ActorTableData.ALIVE`.

File: python/ray/state.py [hunk 3]
```diff
@@ -377,6 +377,54 @@ def profile_table(self):
...
+                return "PENDING"
+            elif state == 1:
+                return "ALIVE"
+            else:
+                return "DEAD"
+
+        def get_strategy(strategy):
+            if strategy == 0:
```
Comment (ffbin) [inline python/ray/state.py:410]: ditto

File: src/ray/protobuf/gcs_service.proto [hunk 4]
```diff
@@ -169,6 +169,9 @@ service PlacementGroupInfoGcsService {
   // Create placement group via gcs service
   rpc CreatePlacementGroup(CreatePlacementGroupRequest)
       returns (CreatePlacementGroupReply);
+
```
Comment (ffbin) [inline src/ray/protobuf/gcs_service.proto:172]: Add a comment?

File: python/ray/experimental/placement_group.py [hunk 5]
```diff
@@ -31,3 +31,10 @@ def placement_group(bundles: List[Dict[str, float]],
         name, bundles, strategy)
 
     return placement_group_id
+
+
+def placement_group_table(placement_group_id):
```
Comment (ericl) [inline python/ray/experimental/placement_group.py:36]: placement_groups()? (for consistency with the other debug calls naming)

**Review by ericl — APPROVED** [review]
Looks good as an internal / experimental API, though I think we'll want to revisit the API definition before moving it out of experimental.

**Comment by rkooo567** [comment]
@ericl Definitely! I think eventually, we should show more information than we just have (for example, scheduling map). I might implement them on the way since it will take only 1~2 hours to implement. We can discuss about the public API in the future.

File: python/ray/gcs_utils.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/gcs_utils.py:2]: Idk why it was changed actually.. My linter just changed it. Do you know how I don't do that? (I am using clang-format7.0 & flake8).

Comment (rkooo567) [reply python/ray/gcs_utils.py:2]: Fixed.

File: python/ray/state.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/state.py:402]: Sounds good. Didn't know we could do that.

File: python/ray/gcs_utils.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/gcs_utils.py:2]: Didn't fix actually. Let me know how I can skip linting. I will just keep it otherwise!

**Review by ffbin — APPROVED** [review]
LGTM

**Comment by ericl** [comment]
clang-format failed

**Comment by rkooo567** [comment]
@ericl It was due to our linter issues (Simon fixed it now). I will merge the PR once it is ready.

**Pull request merged**